nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/actions/CSVExportAction.java

```diff
@@ -0,0 +1,309 @@
+            context.askForCheckpoint();
+        }
+
+    }
+
+    public static class SortBlobLines extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "sortBlobLines";
+
+        public SortBlobLines() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            blob = sort(blob, commandId);
+            if (blob != null) {
+                storeBlob(blob, commandId);
+            }
+            context.produceRecord("o1", BulkRecords.of(commandId, 1, getKey(commandId)));
+            context.askForCheckpoint();
+        }
+
+        // use http://www.algosome.com/articles/how-to-sort-large-file.html
+        protected Blob sort(Blob blob, String commandId) {
+            File file = blob.getFile();
+            if (file == null) {
+                getLog().error(String.format("Blob for %s is not backed by a file, sort is not supported", commandId));
+                return null;
+            }
+            try (Stream<String> lines = Files.lines(file.toPath())) {
+                Path temp = createTemp(commandId);
+                List<String> sorted = lines.sorted().collect(Collectors.toList());
+                Files.write(temp, sorted, StandardOpenOption.WRITE);
+                return new FileBlob(temp.toFile());
+            } catch (IOException e) {
+                getLog().error(e, e);
+                return null;
+            }
+        }
+    }
+
+    public static class ZipBlob extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "zipBlob";
+
+        public ZipBlob() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            try {
+                blob = BlobUtils.zip(blob, null);
+            } catch (IOException e) {
+                getLog().error(e, e);
+            }
+            storeBlob(blob, commandId);
+            context.produceRecord("o1", BulkRecords.of(commandId, countFrom(record), getKey(commandId)));
+            context.askForCheckpoint();
+        }
+
+    }
+
+    public static class ExposeBlobInDownload extends AbstractTransientBlobComputation {
+
+        public static final String NAME = "moveBlobToDownload";
+
+        public ExposeBlobInDownload() {
+            super(NAME);
+        }
+
+        @Override
+        public void processRecord(ComputationContext context, String documentIdsStreamName, Record record) {
+            String commandId = commandIdFrom(record);
+            Blob blob = getBlob(dataFrom(record));
+            if (blob != null) {
```

Comment:
Why this check here and not in `ZipBlob` ?

Reply:
because in ZipBlob the operation is in try-catch but it makes sens to check it too !